File: src/main/java/org/kite9/diagram/common/elements/vertex/CompactionHelperVertex.java
package org.kite9.diagram.common.elements.vertex;

import java.util.Collections;
import java.util.Set;

import org.kite9.diagram.model.DiagramElement;

/**
 * Vertex inserted to aid the compaction process.  Typically added during rectangularization.
 * 
 * @author robmoffat
 *
 */
public class CompactionHelperVertex extends AbstractVertex implements NoElementVertex {

	public CompactionHelperVertex(String name) {
		super(name);
	}

	@Override
	public boolean isPartOf(DiagramElement de) {
		return false;
	}

	@Override
	public Set<DiagramElement> getDiagramElements() {
		return Collections.emptySet();
	}

	@Override
	public DiagramElement getOriginalUnderlying() {
		return null;
	}

}

File: a/src/myMain/kotlin/org/kite9/diagram/common/elements/vertex/CompactionHelperVertex.kt
package org.kite9.diagram.common.elements.vertex

import org.kite9.diagram.model.DiagramElement

/**
 * Vertex inserted to aid the compaction process.  Typically added during rectangularization.
 *
 * @author robmoffat
 */
class CompactionHelperVertex(val id: String) : AbstractVertex(id), NoElementVertex {

    override fun isPartOf(de: DiagramElement?): Boolean {
        return false
    }

    override fun getDiagramElements(): Set<DiagramElement> {
        return emptySet()
    }
}